ansible_rulebook/rule_set_runner.py

```diff
@@ -481,6 +515,43 @@ async def _call_action(
             except BaseException as e:
                 logger.error(e)
                 raise
+        elif action_plugin := self.find_action(action):
+            try:
+                control, action_args, variables_copy = self._build_control(
+                    action,
+                    action_args,
+                    rules_engine_result,
+                    variables,
+                    metadata,
+                    inventory,
+                    hosts,
+                )
+                await action_plugin["main"](metadata, control, **action_args)
+            except Exception as e:
+                logger.error("Error calling action %s, err %s", action, str(e))
+                error = e
+            except BaseException as e:
+                logger.error(e)
+                raise
+        elif has_action(*split_collection_name(action)):
+            action_plugin = find_action(*split_collection_name(action))
+            try:
+                control, action_args, variables_copy = self._build_control(
+                    action,
+                    action_args,
+                    rules_engine_result,
+                    variables,
+                    metadata,
+                    inventory,
+                    hosts,
+                )
+                await action_plugin.main(metadata, control, **action_args)
```

Comment:
@benthomasson Should we pass the metadata, control which are data classes as dictionaries into the action plugin so they can be tested outside of ansible rulebook otherwise they would have to know about the Metadata,Control data class definitions and if we change it it will effect the plugin. Also for the feedback should we be passing in a function that can they call to send the feedback to the eda-server. We should have a very loose coupling between action plugin and ansible-rulebook so each can be tested separately.